include/SDL3/SDL_time.h

```diff
@@ -0,0 +1,244 @@
+ */
+typedef enum SDL_TIME_FORMAT
+{
+    SDL_TIME_FORMAT_24HR = 0, /**< 24 hour time */
+    SDL_TIME_FORMAT_12HR = 1, /**< 12 hour time */
+} SDL_TIME_FORMAT;
+
+/**
+ *  Global date/time properties
+ *
+ *  - `SDL_PROP_GLOBAL_SYSTEM_DATE_FORMAT_NUMBER`: the index of the SDL_DATE_FORMAT enum entry with the
+ *    preferred date display format for the system locale.
+ *  - `SDL_PROP_GLOBAL_SYSTEM_TIME_FORMAT_NUMBER`: the index of the SDL_TIME_FORMAT enum entry with the
+ *    preferred time display format for the current locale.
+ */
+#define SDL_PROP_GLOBAL_SYSTEM_DATE_FORMAT_NUMBER "time.date_format"
+#define SDL_PROP_GLOBAL_SYSTEM_TIME_FORMAT_NUMBER "time.time_format"
+
+/**
+ *
+ * Gets the current value of the system realtime clock in nanoseconds since Jan 1, 1970 UTC.
+ *
+ * \param ticks the SDL_Time to hold the returned tick count
+ * \returns 0 on success or -1 on error; call SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0 
+ */
+extern DECLSPEC int SDLCALL SDL_GetRealtimeClockTicks(SDL_Time *ticks);
+
+/**
+ *
+ * Converts a given SDL_TimeSpec since the epoch to a calendar time, expressed in Universal
+ * Coordinated Time (UTC), in the SDL_DateTime format.
+ *
+ * \param ticks the SDL_Time to be converted
+ * \param dt the resulting SDL_DateTime
+ * \returns 0 on success or -1 on error; call SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC int SDLCALL SDL_TimeToUTCDateTime(SDL_Time ticks, SDL_DateTime *dt);
+
+/**
+ *
+ * Converts a given SDL_TimeSpec since the epoch to a calendar time, expressed in local time,
+ * in the SDL_DateTime format.
+ *
+ * \param ticks the SDL_Time to be converted
+ * \param dt the resulting SDL_DateTime
+ * \returns 0 on success or -1 on error; call SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC int SDLCALL SDL_TimeToLocalDateTime(SDL_Time ticks, SDL_DateTime *dt);
+
+/**
+ *
+ * Converts a calendar time to seconds and nanoseconds since the epoch.
+ * This function ignores the day_of_week member of the SDL_DateTime struct, so it may remain unset.
+ *
+ * \param dt the source SDL_DateTime
+ * \param ticks the resulting SDL_Time
+ * \returns 0 on success or -1 on error; call SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC int SDLCALL SDL_DateTimeToTime(const SDL_DateTime *dt, SDL_Time *ticks);
+
+/**
+ *
+ * Given an SDL_DateTime struct, check if the members are within valid ranges to represent a real date
+ * and time.
+ *
+ * \param dt the SDL_DateTime
+ * \returns SDL_TRUE if all struct members are within valid ranges, otherwise SDL_FALSE; call SDL_GetError()
+ *          for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC SDL_bool SDLCALL SDL_DateTimeIsValid(const SDL_DateTime *dt);
```

Comment:
What's the use case for this?

Reply:
It's currently used internally to make sure that a date/time is valid before trying to convert it. Was thinking that it might be useful externally, but it can probably go, as I haven't really found any use for it.

Reply:
Was removed. Can always be exposed again later if someone has a use for it.